plugins/Monolog/Processor/ExceptionToTextProcessor.php

```diff
@@ -76,17 +81,34 @@ private function getMessage($exception)
 
     private function getStackTrace($exception)
     {
-        if (is_array($exception) && isset($exception['backtrace'])) {
-            return $exception['backtrace'];
-        }
-
         return Log::$debugBacktraceForTests ?: self::getWholeBacktrace($exception);
     }
 
-    public static function getWholeBacktrace(\Exception $exception, $shouldPrintBacktrace = true)
+    /**
+     * @param \Exception|array $exception
+     * @param bool|null $shouldPrintBacktrace
+     * @return mixed|string
+     */
+    public static function getWholeBacktrace($exception, $shouldPrintBacktrace = null)
     {
+        if ($shouldPrintBacktrace === null) {
+            $shouldPrintBacktrace = \Piwik_ShouldPrintBackTraceWithMessage();
+        }
+
+        if (is_array($exception)) {
+            if ($shouldPrintBacktrace && isset($exception['backtrace'])) {
+                $trace = $exception['backtrace'];
+                $trace = self::replaceSensitiveValues($trace);
+                return $trace;
+            } else {
+                return self::BACKTRACE_OMITTED_MESSAGE;
+            }
+        }
+
         if (!$shouldPrintBacktrace) {
-            return $exception->getMessage();
+            $message = $exception->getMessage() . "\n" . self::BACKTRACE_OMITTED_MESSAGE;
```

Comment:
fyi the message again might not be needed? In my case it causes the exception message to be shown twice 
![image](https://user-images.githubusercontent.com/273120/97935823-8b2f2500-1dde-11eb-8668-6c3380c85f8d.png)

At least it happened when I triggered an exception in `Common::safe_unserialize()` and thne it went into the `ExceptionToTextProcessor::__invoke`. I reckon the trace does not need to append the message again

Reply:
I see it might be needed because of eg https://github.com/matomo-org/matomo/pull/16640/files#diff-7aeb867200acd3d0c23432f4b94edae59f9549e2b20160b83bd165830e5e2042L208 

The method name might be bit misleading as it gets the message and the trace?